File: test/fixtures/matlab/post_image.m
%% Web Access using Data Import and Export API
% This is not possible with the webread/webwrite API

%% HTTP Interface
import matlab.net.*
import matlab.net.http.*
import matlab.net.http.io.*

uri = URI('http://localhost:28139/targetservice');
body = MultipartFormProvider('image', ImageProvider('image.jpg'));
response = RequestMessage('post', [], body).send(uri.EncodedURI);

File: test/fixtures/dart/post_image.dart
import 'package:http/http.dart' as http;

void main() async {
  final url = Uri.parse('http://localhost:28139/targetservice');

  final req = http.MultipartRequest('POST', url)
    ..files.add(await http.MultipartFile.fromPath(
      'image', 'image.jpg'));

  final stream = await req.send();
  final res = await http.Response.fromStream(stream);
  final status = res.statusCode;
  if (status != 200) throw Exception('http.send error: statusCode= $status');

  print(res.body);
}
